File: ios/Mobius-Common/Model/RestaurantOrder.swift
import Foundation

public struct OrdersResponse<T: Decodable>: Decodable {
    public let orders: [T]
}

public struct RestaurantOrder: Decodable {
    public let orderID: String
    public let deliveryDate: String
    public let source: String
    public let items: [Item]
    public let totalPrice: Double
    public let currency: String
    public let currentStatus: String

    enum CodingKeys: String, CodingKey {
        case orderID = "order_id"
        case deliveryDate = "delivery_date"
        case source
        case items
        case totalPrice = "total_price"
        case currency
        case currentStatus = "current_status"
    }
}

File: Mobius-Demo-Flutter/lib/feature/order/data/model/restaurant_order.dart
import 'package:meta/meta.dart';

import '../../domain/entity/entity.dart';
import 'item.dart';

@immutable
class RestaurantOrderModel extends RestaurantOrder {
  const RestaurantOrderModel({
    required super.orderID,
    required super.deliveryDate,
    required super.source,
    required super.items,
    required super.totalPrice,
    required super.currency,
    required super.currentStatus,
  });

  factory RestaurantOrderModel.fromMap(Map<String, Object?> map) {
    return RestaurantOrderModel(
      orderID: map['order_id'] as String,
      deliveryDate: map['delivery_date'] as String,
      source: map['source'] as String,
      items: (map['items'] as List<Object?>)
          .map((value) => ItemModel.fromMap(value as Map<String, Object?>))
          .toList(),
      totalPrice: (map['total_price'] as num).toDouble(),
      currency: map['currency'] as String,
      currentStatus: map['current_status'] as String,
    );
  }

  Map<String, Object?> toMap() {
    return {
      'order_id': orderID,
      'delivery_date': deliveryDate,
      'source': source,
      'items': items,
      'total_price': totalPrice,
      'currency': currency,
      'current_status': currentStatus,
    };
  }
}
